mp/src/game/client/clientmode_shared.cpp

```diff
@@ -1116,6 +1116,12 @@ void ClientModeShared::FireGameEvent( IGameEvent *event )
 		{
 			// that's me
 			pPlayer->TeamChange( team );
+
+			auto neoPlayer = dynamic_cast<C_NEO_Player*>(pPlayer);
+			if(neoPlayer)
+			{
+				engine->ClientCmd("classmenu");
+			}
```

Comment:
Wrap our code in `#ifdef NEO`...`#endif` inside Valve blocks.

Also, I don't think you need to do this cast - if we've got a valid player pointer, it should be a safe assertion that it is always a valid `C_NEO_Player*`. Although you could `Assert(dynamic_cast<C_NEO_Player*>(pPlayer))` as a sanity check if you want to, since the assertion would be optimized away from a release build.

But I think since the Valve code at line 1118 above this is calling `pPlayer->TeamChange` here anyway, this might fit better as `virtual void TeamChange(int iNewTeam) override` for the client side `C_NEO_Player` class.